tests/integration_tests/datasources/test_nocloud.py

```diff
@@ -88,3 +95,237 @@ def test_nocloud_seedfrom_vendordata(client: IntegrationInstance):
+            "/server.py",
+            dedent(
+                """\
+                #!/usr/bin/python3
+                import logging
+
+                from pyftpdlib.authorizers import DummyAuthorizer
+                from pyftpdlib.handlers import FTPHandler, TLS_FTPHandler
+                from pyftpdlib.servers import FTPServer
+                from pyftpdlib.filesystems import UnixFilesystem
+
+                encrypted = """
+                + str(encrypted)
+                + """
+
+                logging.basicConfig(level=logging.DEBUG)
+
+                # yeah, it's not secure but that's not the point
+                authorizer = DummyAuthorizer()
+
+                # Define a read-only anonymous user
+                authorizer.add_anonymous("/home/anonymous")
+
+                # Instantiate FTP handler class
+                if not encrypted:
+                    handler = FTPHandler
+                    logging.info("Running unencrypted ftp server")
+                else:
+                    handler = TLS_FTPHandler
+                    handler.certfile = "/cert.pem"
+                    handler.keyfile = "/key.pem"
+                    logging.info("Running encrypted ftp server")
+
+                handler.authorizer = authorizer
+                handler.abstracted_fs = UnixFilesystem
+                server = FTPServer(("localhost", 2121), handler)
+
+                # start the ftp server
+                server.serve_forever()
+                """
+            ),
+        )
+        assert client.execute("chmod +x /server.py").ok
+
+        if encrypted:
+            if CURRENT_RELEASE > FOCAL:
+                assert client.execute("apt install -yq mkcert").ok
+            else:
+
+                # install golang
+                assert client.execute("apt install -yq golang").ok
+
+                # build mkcert from source
+                #
+                # we could check out a tag, but the project hasn't
+                # been updated in 2 years
+                #
+                # instructions from https://github.com/FiloSottile/mkcert
+                assert client.execute(
+                    "git clone https://github.com/FiloSottile/mkcert && "
+                    "cd mkcert && "
+                    "go build -ldflags "
+                    '"-X main.Version=$(git describe --tags)"'
+                ).ok
+
+                # giddyup
+                assert client.execute(
+                    "ln -s $HOME/mkcert/mkcert /usr/local/bin/mkcert"
+                ).ok
+
+            # more palatable than openssl commands
+            assert client.execute(
+                "mkcert -install -cert-file /cert.pem -key-file /key.pem "
+                "localhost 127.0.0.1 0.0.0.0 ::1"
+            ).ok
+
+        client.write_to_file(
+            "/lib/systemd/system/local-ftp.service",
+            dedent(
+                """\
+                [Unit]
+                Description=run a local ftp server against
```

Comment:
I think this was accidentally truncated. Consider:

```
Description=TESTING USE ONLY ftp server
```

or something else that would stand out pretty loudly if someone accidentally runs these tests on their development workstation.